engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/AuthorizationManager.java

```diff
@@ -505,12 +505,39 @@ public void deleteAuthorizationsByResourceIdAndGroupId(Resource resource, String
    */
   public void checkCamundaAdmin() {
     final Authentication currentAuthentication = getCurrentAuthentication();
-    CommandContext commandContext = Context.getCommandContext();
 
-    if (isAuthorizationEnabled() && commandContext.isAuthorizationCheckEnabled()
-        && currentAuthentication != null  && !isCamundaAdmin(currentAuthentication)) {
+    if (isAuthorizationEnabled() && getCommandContext().isAuthorizationCheckEnabled()
+        && currentAuthentication != null && !isCamundaAdmin(currentAuthentication)) {
 
-      throw LOG.requiredCamundaAdminException();
+      throw LOG.requiredCamundaAdminOrPermissionException(null);
```

Comment:
Maybe we can leave the previous method as well and handle reusable parts internally in the logger?